File: javascript/atbash-cipher/atbash-cipher.js
const PLAIN = 'abcdefghijklmnopqrstuvwxyz';
const CIPHER = 'zyxwvutsrqponmlkjihgfedcba';

export const encode = (plainText) => {

  plainText = plainText.toLowerCase().replaceAll(/\W/g, '');

  let encodedText = [];
  for (let i = 0; i < plainText.length; i++) {

    if (i > 0 && i % 5 === 0) {
      encodedText.push(' ');
    }

    encodedText.push(encodeLetter(plainText[i]));
  }

  return encodedText.join('');
};

export const decode = (cipherText) => {

  cipherText = cipherText.replaceAll(/\s+/g, '');

  return cipherText.split('').map(letter => decodeLetter(letter)).join('');

};

const encodeLetter = (letter) => {
  let index = PLAIN.indexOf(letter);

  if (index < 0) {
    return letter;
  }

  return CIPHER.charAt(index);
};

const decodeLetter = (letter) => {
  let index = CIPHER.indexOf(letter);

  if (index < 0) {
    return letter;
  }

  return PLAIN.charAt(index);
};

File: python/atbash-cipher/atbash_cipher.py
import re

PLAIN = "abcdefghijklmnopqrstuvwxyz"
CIPHER = PLAIN[::-1]


def encode(plain_text):
    encoded = "".join(
        CIPHER[PLAIN.index(x)] if x in PLAIN else x
        for x in re.sub(r'\W', "", plain_text).lower())

    return " ".join(encoded[x: x + 5] for x in range(0, len(encoded), 5))


def decode(ciphered_text):
    return "".join(
        PLAIN[CIPHER.index(x)] if x in CIPHER else x
        for x in re.sub(r'\s+', '', ciphered_text))
